src/main/java/alpha/Parser.java

```diff
@@ -0,0 +1,238 @@
+     * @throws InvalidFormatException If /from and /to are not found within the input.
+     */
+    public Command addEvent(String input) throws MissingInfoException, InvalidFormatException {
+        String[] splitInput = input.split(" ");
+        if (splitInput.length < 3) {
+            throw new MissingInfoException("Missing Information!", TaskException.TaskType.EVENT);
+        } else {
+            String[] splitEvent = input.split("/");
+            if (splitEvent[1].startsWith("from") && splitEvent[2].startsWith("to")) {
+                Event event = Event.makeEvent(splitEvent[0].substring(6),
+                        splitEvent[1].substring(5),
+                        splitEvent[2].substring(3));
+                return new AddCommand(taskList, fileHandler, ui, event);
+            } else {
+                throw new InvalidFormatException("Invalid Format!", TaskException.TaskType.EVENT);
+            }
+        }
+    }
+
+    /**
+     * Returns a Command that adds a ToDo based on an input.
+     *
+     * @return a Command that adds a ToDo.
+     * @throws MissingInfoException If length of splitInput == 1
+     */
+    public Command addToDo(String input) throws MissingInfoException {
+        String[] splitInput = input.split(" ");
+        if (splitInput.length == 1) {
+            throw new MissingInfoException("Missing Information!", TaskException.TaskType.TODO);
+        } else {
+            ToDo todo = ToDo.createToDo(input.substring(5));
+            return new AddCommand(taskList, fileHandler, ui, todo);
+        }
+    }
+
+    /**
+     * Returns a Command that adds a deadline based on an input.
+     *
+     * @return a Command that adds a ToDo.
+     * @throws MissingInfoException If length of splitInput == 1
+     */
+    public Command addDeadline(String input) throws MissingInfoException, InvalidFormatException, DateTimeParseException {
+        String[] splitInput = input.split(" ");
+        if (splitInput.length < 2) {
+            throw new MissingInfoException("Missing Information!", TaskException.TaskType.DEADLINE);
+        } else if (input.split("/by").length != 2) {
+            throw new InvalidFormatException("Invalid Format!", TaskException.TaskType.DEADLINE);
+        } else {
+            String[] splitDeadline = input.split("/by");
+            Deadline deadline = Deadline.makeDeadline(splitDeadline[0].substring(9),
+                    splitDeadline[1]);
+            return new AddCommand(taskList, fileHandler, ui, deadline);
+        }
+    }
+
+    /**
+     * Returns a Command that deletes a task from storage based on an input.
+     *
+     * @return a Command that deletes a task.
+     * @throws MissingIndexException if the index is missing.
+     * @throws InvalidIndexException if the index exceeds or is lesser than the number of stored tasks.
+     */
+    public Command delete(String input) throws MissingIndexException, InvalidIndexException {
```

Comment:
perhaps a more specific name for the methods, and for the other methods in several other places as well